backend/modules/visitorActivity/daos/visitor.activity.s3.dao.ts

```diff
@@ -0,0 +1,53 @@
+import AWS from 'aws-sdk';
+import debug from "debug";
+import { config } from 'dotenv';
+
+const log: debug.IDebugger = debug("app:in-memory-dao");
+
+class VisitorActivityS3Dao {
+
+  s3 = new AWS.S3({
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY
+  });
+
+  constructor() {
+    log("Created new instance of VisitorActivityS3Dao", process.env.AWS_BUCKET_NAME);
+    config();
+  }
+
+  async uploadActivityEvents(events: any[]) {
+    try {
+      const filename = Date.now();
```

Comment:
any logic moveed into private function, what the data should be apparent upon visiting the helper private function used, otherwise,, dao should just take data, prcess data through a private function and return the query result